lib/optimizely/project_config.rb

```diff
@@ -120,13 +120,13 @@ def initialize(datafile, logger, error_handler)
         end
       end
       @rollout_id_map = generate_key_map(@rollouts, 'id')
-      # split out the experiment id map for rollouts
-      @rollout_experiment_id_map = {}
+      # split out the experiment key map for rollouts
+      @rollout_experiment_key_map = {}
       @rollout_id_map.each_value do |rollout|
         exps = rollout.fetch('experiments')
-        @rollout_experiment_id_map = @rollout_experiment_id_map.merge(generate_key_map(exps, 'id'))
+        @rollout_experiment_key_map = @rollout_experiment_key_map.merge(generate_key_map(exps, 'key'))
       end
-      @all_experiments = @experiment_key_map.merge(@rollout_experiment_id_map)
```

Comment:
Wow this is a big mistake we didn't catch in the code review last time around

Reply:
Luckily IDs and keys have been same, but if it had changed things would be really very bad.